client/src/pages/Job/index.tsx

```diff
@@ -1,25 +1,71 @@
-import JobDetails from '../../components/jobDetails';
-import ProposalForm from '../../components/ProposalForm';
-import { JobAbout } from '../../interfaces';
+import axios from 'axios';
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { CircularProgress } from '@mui/material';
+import './style.css';
+import { Client, JobAboutPage } from '../../interfaces';
+import { JobDetails, ProposalForm } from '../../components';
 
-const job: JobAbout = {
-  title: 'Graphic Designer',
-  description: `Lorem ipsum dolor sit amet consectetur 
-          adipisicing elit. Doloribus deserunt provident
-          adipisci dignissimos aliquid, enim nesciunt.`,
-  category: 'Design',
-  duration: '2day',
-  budget: 20,
+function Job() {
+  const { id } = useParams();
+  const [jobState, setJob] = useState<JobAboutPage>({
+    time: '',
+    title: '',
+    description: '',
+    category: '',
+    budget: 0,
+    user: {
+      name: '', email: '', id: 0, role: '',
+    },
+  });
+  const [client, setClient] = useState<Client>({
+    name: '', email: '', id: 0, role: '',
+  });
+  const [error, setError] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(false);
 
-};
+  const handleClick = () => {
+    window.scrollTo(0, 500);
+  };
+
+  useEffect(() => {
+    const getJobData = async () => {
+      setLoading(true);
+      setError(false);
+      try {
+        const jobData = await axios.get(`/api/v1/jobs/${id}`);
+        setLoading(false);
+        setJob(jobData.data.data);
+        setClient(jobData.data.data.user);
+      } catch (err) {
+        setLoading(false);
+        setError(true);
+      }
+    };
+    getJobData();
+  }, []);
+
+  if (loading) {
+    return (
+      <div className="spinner">
+        <CircularProgress color="inherit" />
+        {' '}
+      </div>
+    );
+  }
+
+  if (error) {
+    return (
+      <h2 className="error">Job Not Found</h2>
+    );
+  }
 
-function Job() {
   return (
     <div className="container">
-      <JobDetails job={job} />
+      {jobState && <JobDetails job={jobState} client={client} handleClick={handleClick} /> }
```

Comment:
```suggestion
      {<JobDetails job={jobState} client={client} handleClick={handleClick} /> }
```
NO need as the initial state it's not null anymore